### PR TITLE
Added initial support for ZR300 camera

### PR DESCRIPTION
**Pull request by reaganlo**
**Please ensure the above _guidelines for contributing_ are met.**

Fixes Issue: 
Added initial support for ZR300 camera

Changes proposed in this pull request: 
- Updated code for librealsense v1.9.7 compatibility.
- Added optical frame ids for IR and IR2 streams to keep it consistent with the Depth and Color streams.
- Added Fisheye stream for ZR300.
- Added IMU capability for ZR300.
- Added basic unit tests for Fisheye and IMU data
- Update README with ZR300 details and erratum.

### CODE REVIEW

**Comment by mdhorn** [comment]
@reaganlo You should rebase this PR.

File: realsense_camera/README.md [hunk 1]
```diff
@@ -324,3 +351,5 @@ Until it gets fixed, use a single launch file for multiple F200 or SR300 cameras
 [Issue #89](https://github.com/intel-ros/realsense/issues/89).
 Unlikely to be fixed as current plan is to
 [remove native point cloud generation](https://github.com/intel-ros/realsense/issues/47).
+
+* The service "get_imu_info" is yet to be tested.
```
Comment (kevincwells) [inline realsense_camera/README.md]: Why are we introducing a service if we haven't tested it? Or is this comment out of date?

File: realsense_camera/cfg/zr300_params.cfg [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+gen.add("color_saturation",                                int_t,    0,     "Saturation",                                 64,        0,      255)
+gen.add("color_sharpness",                                 int_t,    0,     "Sharpness",                                  50,        0,      7)
+
+# Must be set only if color_enable_auto_white_balance is disabled
+gen.add("color_white_balance",                             int_t,    0,     "White Balance",                              4600,      2000,   8000)
+
+gen.add("color_enable_auto_exposure",                      int_t,    0,     "Enable Auto Exposure",                       1,         0,      1)
+gen.add("color_enable_auto_white_balance",                 int_t,    0,     "Enable Auto White Balance",                  1,         0,      1)
+gen.add("r200_lr_auto_exposure_enabled",                   int_t,    0,     "LR Auto Exposure Enabled",                   0,         0,      1)
+gen.add("r200_lr_gain",                                    int_t,    0,     "LR Gain",                                    400,       100,    6399)
```
Comment (kevincwells) [inline realsense_camera/cfg/zr300_params.cfg:27]: If all of these parameters are generic or used in multiple cameras, they should probably not have "r200" in the name.

**Comment by rjingar** [comment]
Looks good to me. Approved 👍

File: realsense_camera/include/realsense_camera/base_nodelet.h [hunk 3]
```diff
@@ -116,14 +117,13 @@ namespace realsense_camera
     int cv_type_[STREAM_COUNT];
     int unit_step_size_[STREAM_COUNT];
     int step_[STREAM_COUNT];
-    int ts_[STREAM_COUNT];
+    double ts_[STREAM_COUNT];
```
Comment (kevincwells) [inline realsense_camera/include/realsense_camera/base_nodelet.h:120]: Wow, did they really change timestamps to be doubles?!

**Review by kevincwells — APPROVED** [review]
This looks good.  I made a comment about naming, but that's minor.

File: realsense_camera/README.md (hunk 1 above)
Comment (reaganlo) [reply realsense_camera/README.md]: @kevincwells Removed the comment. Was able to test the service.

File: realsense_camera/cfg/zr300_params.cfg (hunk 2 above)
Comment (reaganlo) [reply realsense_camera/cfg/zr300_params.cfg:27]: @kevincwells As discussed, even if we remove the term "r200" from the dynamic config, it still appears when we read them directly using the get_settings service. Instead, will raise an issue with librealsense to rename these camera options and then make the change in the dynamic config.

**Pull request merged**